magma/fromverilog.py

```diff
@@ -130,23 +130,17 @@ def FromVerilog(source, func, type_map, module=None):
                 # inline source
                 circuit.verilogFile = source
                 EndDefine()
-            if module is not None:
-                assert node.name == module
-                return circuit
             modules.append(circuit)
         except Exception as e:
             logger.warning(f"Could not parse module {node.name} ({e}), "
                            f"skipping")
-    if module is not None:
-        raise Exception(f"Could not find module {module}")
-
```

Comment:
This changes the behavior to return an empty list if there is no module found. Seems like reasonable behavior, but before it would raise an error if it couldn't find the module.

Either behavior seems like it could work, but maybe raising an exception if `target_modules` is not empty but no modules were found, might be helpful for the user (forcing them to handle this case, rather than returning an empty list which might make their code advance and cause a bug later down the line that may be harder to understand).

Reply:
Understood. Perhaps a warning would be best? The information is ultimately conveyed one way or another when an empty list is returned.

Reply:
warning sounds good, so if there is some issue caused, they can at least inspect the log and see it which might give them a clue as to where the issue is coming from